graphbolt/src/random_engine.h

```diff
@@ -0,0 +1,84 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file random_engine.h
+ * @brief Random engine used for sampling.
+ */
+
+#ifndef GRAPHBOLT_RANDOM_ENGINE_H_
+#define GRAPHBOLT_RANDOM_ENGINE_H_
+
+#include <random>
+
+namespace graphbolt {
+namespace sampling {
+
+class RandomEngine {
+ public:
+  // Get the instance of the RandomEngine (Singleton).
```

Comment:
// Singletons should not be cloned or assigned.
    RandomEngine(RandomEngine &other) = delete;
    void operator=(const RandomEngine &) = delete;